app/models/push.rb

```diff
@@ -0,0 +1,201 @@
+class Push < ApplicationRecord
+  enum :kind, [:text, :file, :url], default: :text
+
+  validates :kind, inclusion: { in: self.kinds, message: "%{value} is not a valid push type" }, presence: true
+  validate :check_enabled_push_kinds
+  validates :url_token, presence: true, uniqueness: true
+
+  with_options on: :create do |create|
```

Comment:
Previously, some checks or assignments are done on different places. I have put all them into one place. These checks and assignments will be run when a Push instance is created on the console, too. It prevents to avoid to save incomplete data.

Reply:
Let me remind [the order of active records callbacks](https://guides.rubyonrails.org/active_record_callbacks.html#creating-an-object) when a record is created;

  before_validation
  after_validation
  before_save
  around_save
  before_create
  around_create
  after_create
  after_save
  after_commit / after_rollback